docs/USAGE-GUIDE.md

```diff
@@ -1142,18 +1142,24 @@ object.
 ### s2n\_connection\_set\_fd
 
 ```c
-int s2n_connection_set_fd(struct s2n_connection *conn, 
+int s2n_connection_set_fd(struct s2n_connection *conn,
                           int readfd);
-int s2n_connection_set_read_fd(struct s2n_connection *conn, 
+int s2n_connection_set_read_fd(struct s2n_connection *conn,
                                int readfd);
-int s2n_connection_set_write_fd(struct s2n_connection *conn, 
+int s2n_connection_set_write_fd(struct s2n_connection *conn,
                                 int writefd);
 ```
 
 **s2n_connection_set_fd** sets the file-descriptor for an s2n-tls connection. This
-file-descriptor should be active and connected. s2n-tls also supports setting the
-read and write file-descriptors to different values (for pipes or other unusual
-types of I/O).
+file-descriptor should be active and connected; one process is reading from a pipe
+before another writes to it. s2n-tls also supports setting the read and write
+file-descriptors to different values (for pipes or other unusual types of I/O).
+
+**Important Note:**
+If the reading process fails to start or is terminated unexpectedly, writing
+to the pipe will raise a SIGPIPE signal. A SIGPIPE signal will cause the process to
```

Comment:
This isn't completely accurate, since it's the pipe we care about and not the process.
```suggestion
If the read end of a pipe is closed unexpectedly, writing
to the pipe will raise a SIGPIPE signal. A SIGPIPE signal will cause the process to
```